src/viam/sdk/spatialmath/orientation.cpp

```diff
@@ -107,7 +107,7 @@ proto::Orientation OrientationConfig::to_proto() const {
             aa.set_y(a.y);
             aa.set_z(a.z);
             aa.set_theta(a.theta);
-            orientation.set_allocated_axis_angles(&aa);
+            *orientation.mutable_axis_angles() = aa;
```

Comment:
You can likely ` = std::move(aa)` here, and similar below.

Reply:
I _think_ @acmorrow is suggesting that you do `*orientation.mutable_axis_angles() = std::move(aa);`. I don't know if there's a big intrinsic difference between `set_allocated_axis_angles` vs `mutable_axis_angles() =`, but, from what I can tell, it's a good habit to move values around instead of accidentally copying them. What you have currently may copy the `proto::Orientation_AxisAngles` object, so I think Drew's suggesting you explicitly move the value. I believe what you had before was setting the `axis_angles` field to an rvalue, which will _also_ not copy the object if I'm not mistaken.

Reply:
Yeah, I didn't even look so carefully. I suspect that using the `set_allocated` wasn't actually correct to begin with (and this file is the only file in the SDK that uses it), because I think it means that the proto takes ownership of the pointed to object, but here that's on the stack (!).

But all I was noting was that `*... = aa` is going to be a copy, but `*... = std::move(aa)` *might* be able to move, and will degrade to a copy if it can't, for some reason, so better to write it that way.

I'm happy with the code that is here now.